### PR TITLE
DOCS: Add release checklists

### PR DESCRIPTION
**Pull request by unclecheese**
* Releases are not initiated by creating an issue, based on one of three templates
* Adds minor, patch, and security release checklists as issue templates
* Update docs to remove deprecated "archive" and "upload" steps
* Remove documentation that reiterates what is in the checklists

## y so much duplication dude

If markdown allowed us to do `<% include %>`, this wouldn't be a problem, but the fact is, we follow different paths for patch, minor, and security releases, which means we can either:

* have three different interdependent checklists that reference each other with hyperlinks (annoying)
* create a bunch of if/else blocks in the checklist (defeats the purpose of a checklist)
* Or.. duplicate them and accept the overhead of making multiple updates when they do change. 

Since the release process isn't something that should be changing often, the last option feels the best fit, as opposed to having the release manager keep three different browser tabs open with partially complete checklists.

But I'm open to ideas!

## Dependencies
- [ ] https://github.com/silverstripe/silverstripe-installer/pull/264

### CODE REVIEW

**Comment by ScopeyNZ** [comment]
Won't this give any user the ability to create one of these checklists when raising an issue on this repo? Perhaps this should exist on the `installer` repo or something where raising issues is less likely? Or just not exist in the `.github/ISSUE_TEMPLATES` folder at all?

**Comment by unclecheese** [comment]
Screenshot 2019-09-24 11 03 46

**Comment by unclecheese** [comment]
We've had the process of how to do a core release documented publicly for years on `docs.silverstripe.org`, so I fail to see how giving users access to create/modify the checklist is any different than what we have right now. We're pretty open about how we do releases, and if someone wants to open a release issue nefariously, then we can just close it.

I thought about doing it on installer, because that could keep things more tidy, but framework has always been our one-stop-shop for issue creation and meta content like docs, so this felt like a better fit. I think as long as our standard issue template takes a new title and description for "Report a bug" and it appears first in the list, it's not a huge change.

**Comment by ScopeyNZ** [comment]
Yeah my concern is that although users probably never browse to the "making a core release" part of the docs (which is completely irrelevant if you're not a core contributor) they can't avoid the big green button next to 3 individual options when they click the "new issue" button for framework.

Users shouldn't even really see the option imo - especially when "Make a feature request" isn't even an option.

File: .github/ISSUE_TEMPLATE/RELEASE_SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  - [ ] Deploy
+- [ ] docs.silverstripe.org updated:
+  - [ ] New changelog showing
+  - [ ] Links to changelogs in announcements work
+- [ ] Announce in the ["Releases" forum](https://forum.silverstripe.org/c/releases)
+- [ ] #ss4 Slack channel topic shows newest release with link to changelog
+- [ ] Security release pages published
+- [ ] Response to issue reporter with reference to the release on the same discussion thread (cc security@silverstripe.org) sent
+- [ ] [CVE publication request](https://cveform.mitre.org/) submitted with link to disclosure on silverstripe.org
+- [ ] Comms to OS teams and marketing
```
Comment (unclecheese) [inline .github/ISSUE_TEMPLATE/RELEASE_SECURITY.md]: Add: close out sercurity issues

**Comment by unclecheese** [comment]
Per the feedback, I've moved the checklists to `silverstripe/installer`.

**Comment by brynwhyman** [comment]
A little late to the party, but these templates might make more sense in https://github.com/silverstripeltd/product-issues. I'm not sure that all core committers have access to that repository but these checklists seems pretty internal (SSltd) to me, i.e 'let Marketing know'. As nice as it would be, I highly doubt anyone other than an internal employee will be following all of the steps in these checklists.

I'd say the process as described on docs.silverstripe.org should be stay suitable for those who are interested our release process or core committers?

Plus, we might want to update these checklists slightly (after merge) to reference CWP releases, which doesn't _need_ to be public.

**Comment by unclecheese** [comment]
In practice, it is always an SS Ltd employee doing the releases, but to concede that it will always be is a bit antithetical to our goals as a provider of open source software, IMO. 

But in any event, it's not so much about who is working the checklists so much as the transparency around:

* How we do our releases
* Where we are in the process of a given release

It's entirely conceivable that someone in the community would have insight around how we could improve these processes. Currently, that's in the documentation, but I don't think it belongs there. It's just not useful when you're in the middle of a release to be reading through paragraphs of self-referential information. A punchlist is much easier to use, and part of this PR is to migrate the "todo" part of the release into an actual todo list.

We also have stakeholders who are waiting on security or patch releases, and having the transparency of how close we are to being finished with a release I think could be valuable.

In general, "open by default" is a good principle to follow, and in my view we have no reason to obscure this.

**Comment by brynwhyman** [comment]
My main thought for moving a release checklist to product-issues is to take into account that a release may not only involve 'silverstripe-installer' and at this stage, we're trying to move further _towards_ coupled releases of all our commercially supported modules.

I don't think it's a problem that this information is public. I'm also all for having information on how we release available for everyone to view and comment on. It's just that it's not being maintained or captured in a place central to our release process.

Aside, I think the checklist is a great idea and much better than our previous 'copy and paste' method of the previous release issue!

**Comment by chillu** [comment]
My 2cts: Module releases could conceivably be performed by open source community members as well (core committers), and we should have a clear public process for those. This could include "core recipe" releases. 

On the other hand, some recipe releases (CWP, the upcoming "essentials" recipe) are directly tied to guarantees Silverstripe makes towards their customers. So those are fine to keep in private repos. 

I'm not sure how feasible it is to keep those as separate processes though, sounds like lots of duplication? If in doubt, I'd default to working out in the open.

File: docs/en/05_Contributing/05_Making_A_SilverStripe_Core_Release.md [hunk 2]
```diff
@@ -88,6 +81,18 @@ For doing security releases the following additional setup tasks are necessary:
   and the [silverstripe.org CMS](http://www.silverstripe.org/admin).
 * Permission on [security pre-announcement mailing list](https://groups.google.com/a/silverstripe.com/forum/#!forum/security-preannounce).
 
+## Initiating a core release
+
+Releases are documented as issues in the [silverstripe/framework Github repository](https://github.com/silverstripe/silverstripe-framework). Create an issue
```
Comment (brynwhyman) [inline docs/en/05_Contributing/05_Making_A_SilverStripe_Core_Release.md]:
```suggestion
Releases are documented as issues in the [silverstripe/silverstripe-installer Github repository](https://github.com/silverstripe/silverstripe-installer). Create an issue
```

As per the dependant PR, these issue templates have moved.

**Comment by chillu** [comment]
Looking through the actual PR, I think this falls short on what I envisioned with this originally. Which is unsurprising since I didn't define any ACs :D The PR moves the current core release docs into a checklist, which is a good first step. But realistically, a core release is virtually always tied to one or more "recipe releases" as well. And the main value is in formalising that recipe release process, which is currently not in checklist form (see https://silverstripe.atlassian.net/wiki/spaces/CWP/pages/655458662/CWP+Quarterly+Release+Process).

Proposed ACs:

 * Core and CWP recipe releases can be created collaboratively by Silverstripe staff members
 * The steps and their order is clearly defined
 * The process enables different recipe release variations: Minor, patch and security
 * The process enables releasing a Core recipe without a full CWP recipe
 * There is a record which steps are completed, when they were completed, and who completed them
 * The process is maintainable (minimises duplication)

Notes:

 * The more steps we automate, the less we need to document, but that's out of scope for this issue
 * There's no mention of all this being public. As much as I'd like that to be the case, there CWP recipe release process should be kept private. If we can make the Core recipe release process a clean subset of this without causing confusion, that's fine. But I'd err on the side of consistent tooling here, and make that checklist private as well. As a compromise, we could summarise the process on docs.silverstripe.org, rather than publish the steps directly.

**Comment by emteknetnz** [comment]
This pull request hasn't had any activity for a while. Are you going to be doing further work on it, or would you prefer to close it now?

**Comment by brynwhyman** [comment]
This template was trialed for a release in April 2020 and has since made its way to being an issue template in silverstripeltd/product-issues: https://github.com/silverstripeltd/product-issues/blob/master/.github/ISSUE_TEMPLATE/RELEASE_CHECKLIST.md

That's being iterated on until we can get to decoupling the full release process that's outlined here (https://silverstripe.atlassian.net/wiki/spaces/CWP/pages/655458662/CWP+Quarterly+Release+Process) from CWP and allow it to be followed for any recipe release. There are a lot of QA steps in that process that are still missing from the checklist in [silverstripeltd/product-issues](https://github.com/silverstripeltd/product-issues/blob/master/.github/ISSUE_TEMPLATE/RELEASE_CHECKLIST.md)

**Comment by chillu** [comment]
@brynwhyman @steve-silverstripe I think that ball is in your court. By creating a release checklist in an access protected repo, you've decided that only SS Ltd staff will be in charge of releases. As per my comments above, this might be a necessary evil, but looking at the current [checklist](https://github.com/silverstripeltd/product-issues/blob/master/.github/ISSUE_TEMPLATE/RELEASE_CHECKLIST.md), this feels like something that could also live on an OSS repo? Either way, this is now a *parallel* checklist to the one we have on https://docs.silverstripe.org/en/4/contributing/making_a_silverstripe_core_release/. This needs to be resolved, these docs aren't accurate any more.

Option A: Extract detailed process from https://docs.silverstripe.org/en/4/contributing/making_a_silverstripe_core_release/ into a new core release checklist as a framework issue template. Accept that some steps might be limited to staff members for now (e.g. "test on SC environment")
Option B: Describe current process on https://docs.silverstripe.org/en/4/contributing/making_a_silverstripe_core_release/ (and state that only SS Ltd staff has access to the actual core release process). Remove all bits which aren't actually followed any more from those docs

My vote is for Option A.

**Comment by GuySartorelli** [comment]
Note: This should not be merged here. I have started [the process of migrating docs](https://github.com/silverstripe/silverstripe-framework/issues/10215) to [a new repository](https://github.com/silverstripe/doc) so this PR (if still relevant) should be closed and a new one should be created there.

**Comment by GuySartorelli** [comment]
Closing as this is unlikely to be updated and is mostly in the wrong place now that docs have moved anyway.

**Pull request closed without merging**